packages/blade/src/components/Avatar/Avatar.web.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import type { AvatarProps } from './types';
+import { StyledAvatar } from './StyledAvatar';
+import { DefaultAvatarIcon } from './DefaultAvatarIcon';
+import { useAvatarGroupContext } from './AvatarGroupContext';
+import { getStyledProps } from '~components/Box/styledProps';
+import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+import { useDropdown } from '~components/Dropdown/useDropdown';
+import BaseButton from '~components/Button/BaseButton/BaseButton';
+import { throwBladeError } from '~utils/logger';
+import type { BladeElementRef } from '~utils/types';
+import { DropdownButton } from '~components/Dropdown';
+
+const getInitials = (name: string): string => {
+  // Combine first and last name initials
+  const names = name.trim().split(' ');
+  if (names.length === 1) {
+    return name.substring(0, 2);
+  }
+  return names[0].substring(0, 1) + names[names.length - 1].substring(0, 1);
+};
+
+const _Avatar: React.ForwardRefRenderFunction<BladeElementRef, AvatarProps> = (
+  {
+    name,
+    color = 'neutral',
+    size = 'xsmall',
+    variant = 'circle',
+    icon,
+    href,
+    target,
+    rel,
+    // Image Props
+    src,
+    alt,
+    srcSet,
+    crossOrigin,
+    referrerPolicy,
+    testID,
+    ...styledProps
+  },
+  ref,
+) => {
+  if (src && !alt && !name) {
+    throwBladeError({
+      moduleName: 'Avatar',
+      message: '"alt" or "name" prop is required when the "src" prop is provided.',
+    });
+  }
+  const groupProps = useAvatarGroupContext();
+  const avatarSize = groupProps?.size ?? size;
+
+  const { dropdownTriggerer } = useDropdown();
```

Comment:
I feel we should have DropdownAvatar kind of component like DropdownButton so the Dropdown bundle size moves into that component only and Avatar stays small